library/ssl_tls13_server.c

```diff
@@ -1054,6 +1056,129 @@ static int ssl_tls13_write_server_hello( mbedtls_ssl_context *ssl )
     return( ret );
 }
 
+/*
+ * Handler for MBEDTLS_SSL_ENCRYPTED_EXTENSIONS
+ */
+static int ssl_tls13_prepare_encrypted_extensions( mbedtls_ssl_context *ssl )
```

Comment:
We should have the same structure as in `ssl_tls13_postprocess_server_hello` I'd say. A `cleanup` label where we zeroize `traffic_keys` and free `transform_handshake` in case of error. Jump to `cleanup` in case of any error.

Reply:
Should we move all key generation codes to server hello ?
And should we do it in this PR ?

Reply:
It would be indeed better to set the handshake transform at the same place in the client and server code thus after the parsing/writing of the server hello. We could also factorize the common code into an `mbedtls_ssl_tls13_set_handshake_transform()` function in ssl_tls13_keys.c. We can do this in this PR or in a specific one, as you prefer.

Reply:
done， please review